src/NUnitFramework/FrameworkVersion.cs

```diff
@@ -1,9 +1,9 @@
 // Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
 
-using System.Reflection;
+//using System.Reflection;
 
 //
 // Current version for the NUnit Framework
 //
-[assembly: AssemblyVersion("4.0.1.0")]
-[assembly: AssemblyFileVersion("4.0.1.0")]
+//[assembly: AssemblyVersion("4.0.1.0")]
```

Comment:
Is removing this not enough?
We pass _version_ to MSBuild

Reply:
In order for MSBuild to override the version the GenerateAssemblyInfo has to be set to true.  If set to false, the msbuild properties are ignored.  
If set to true, we must move all other assembly instructiosn that today are in the CommonAssemblyInfo.cs file into the csproj files.  Those that don't vary with framework could go into the direcoty.build.props file, but Configuration has to go into every csproj we have, since they run on different sets of target frameworks.  
That is doable, but the approach of simply generating a single file from the cake script seemed like an easier approach.

Reply:
@manfred-brands I've been thinking, and realize I did a "Design by surprise" here :-(  .  Flat on my face here.  And thanks for the kick!  ( I just tend to like code generation. ) 
So, I removed the code generation, added conditionals into directory.msbuild.props (perhaps a a bit too many, but they dont do any harm, and then i avoided  more compiler errors).  I then also removed most of the assemblyinfo.cs information and moved that to the csproj files.
And voila, all builds as it should, versions are correct and everything looks like normal again.